docs(Cockpit): add new doc how-to on sending metrics / logs to Cockpit with OTLP (#6663)
scaleway/docs-content: pages/cockpit/how-to/send-metrics-logs-to-cockpit-with-otlp.mdx

---

> title: Send metrics or logs to Cockpit using OTLP

Reviewer: ```suggestion
title: How to send metrics or logs to Cockpit using OTLP
```

description: Learn how to send metrics or logs to Cockpit using OTLP exporters, either with a vanilla OTLP exporter or Grafana Alloy.
tags: cockpit observability otlp metrics logs push-metrics push-logs
dates:
  validation: 2026-07-06
  posted: 2026-07-06
---
import Requirements from '@macros/iam/requirements.mdx'

> This guide explains how to configure OTLP (OpenTelemetry Protocol) to push your metrics and logs to Cockpit. You can choose between using a vanilla OTLP exporter or Grafana Alloy as your metrics collector and forwarder.

Reviewer: ```suggestion
This guide explains how to configure OpenTelemetry Protocol (OTLP) to push your metrics and logs to Cockpit. You can choose between using a vanilla OTLP exporter or Grafana Alloy as your metrics collector and forwarder.
```

<Message type="note">
  Cockpit supports both metrics and logs ingestion via OTLP. Use the following endpoints based on your data source region:
  - **Metrics endpoint**: `https://<your-resource-id>.metrics.cockpit.<region>.scw.cloud/otlp/v1/metrics`
  - **Logs endpoint**: `https://<your-resource-id>.logs.cockpit.<region>.scw.cloud/otlp/v1/logs`

  Retrieve your endpoint URLs from the [Data sources tab](https://console.scaleway.com/cockpit/dataSource) of the Scaleway console.

> </Message>

Reviewer 2: ```suggestion

```

Reviewer 2: I would put this Note under `<Requirements />` (see below).

<Requirements />

  - [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
  - A Scaleway account logged into the [console](https://console.scaleway.com)
  - A [token](/cockpit/how-to/create-token/) with push permissions for metrics and/or logs in the **same region as your data**

>   - For vanilla OTLP: An OTLP-compatible exporter installed on your system

Reviewer 2: ```suggestion
  - For vanilla OTLP: 
    - An OTLP-compatible exporter installed on your system (e.g., OpenTelemetry Collector, otelcol)
    - Your metrics or logs endpoint URLs retrieved from the [Data sources tab](https://console.scaleway.com/cockpit/dataSource) of the Scaleway console

    <Message type="note">
    Cockpit supports both metrics and logs ingestion via OTLP. Use the following endpoints based on your data source region:
    - **Metrics endpoint**: `https://<your-resource-id>.metrics.cockpit.<region>.scw.cloud/otlp/v1/metrics`
    - **Logs endpoint**: `https://<your-resource-id>.logs.cockpit.<region>.scw.cloud/otlp/v1/logs`
    </Message>

```

Reviewer 2: I moved the Note here, it feels more relevant here.

  - For Grafana Alloy: [Installed Grafana Alloy](https://grafana.com/docs/alloy/latest/get-started/install/)

<Message type="important">

>  - Having the default configuration on your agents might lead to more of your resources' metrics being sent, resulting in high consumption and increased costs.

Reviewer 2: ```suggestion
 - If you keep the agents at their default configuration, they may transmit a larger set of resource metrics, which can increase consumption and drive up costs.
```

 - Sending metrics and logs for Scaleway resources or personal data using an external path is a billable feature. Any data that you push yourself is billed, even if you send data from Scaleway products. Refer to the [product pricing](https://www.scaleway.com/en/pricing/managed-services/#cockpit) for more information.
</Message>

## Using a Vanilla OTLP Exporter

This section demonstrates how to configure a standard OTLP exporter to send metrics directly to Cockpit.

### Prerequisites

- An OTLP-compatible exporter (e.g., OpenTelemetry Collector, otelcol)
- Your metrics and/or logs endpoint URLs from the [Data sources tab](https://console.scaleway.com/cockpit/dataSource)

> - A Cockpit token with push permissions

Reviewer 2: ```suggestion
```

Reviewer 2: I would remove this, because we have the list of Requirements above (I extended the list of Requirements above so that it is aligned with this one).

> ### Configuration

Reviewer 2: ```suggestion
```

Reviewer 2: Not needed anymore if we remove Prerequisites.

       headers:
         X-TOKEN: "COCKPIT_TOKEN_SECRET_KEY"
     otlphttp/cockpit-logs:
       endpoint: "https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.logs.cockpit.fr-par.scw.cloud/otlp/v1/logs"
       headers:
         X-TOKEN: "COCKPIT_TOKEN_SECRET_KEY"

   service:
     pipelines:
       metrics:
         receivers: [otlp]
         processors: [batch]
         exporters: [otlphttp/cockpit-metrics]
       logs:
         receivers: [otlp]
         processors: [batch]
         exporters: [otlphttp/cockpit-logs]
   ```

3. Replace the following values:
   - `https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.metrics.cockpit.fr-par.scw.cloud/otlp/v1/metrics` with your metrics endpoint URL
   - `https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.logs.cockpit.fr-par.scw.cloud/otlp/v1/logs` with your logs endpoint URL (optional, if sending logs)
   - `COCKPIT_TOKEN_SECRET_KEY` with your Cockpit token's secret key

   <Message type="important">
     Make sure that your token and your data sources (metrics and/or logs) were both created in the same region.
   </Message>

   <Message type="note">

>      You can configure both metrics and logs pipelines to send data to Cockpit. Simply uncomment or add the logs pipeline configuration if you want to send logs. Each data type requires its own endpoint:

Reviewer 2: I think saying "uncomment" is not correct here. At least, I can't see any commented out lines in the config file provided above. So I can see that we have a `pipelines` section in the example config file, but I'm not sure I understand what the user has to do to "activate/enable" the logs pipeline.

Author: Nice catch, I will remove this, I think it's better to have both metrics and logs in the example

     - Metrics: `/otlp/v1/metrics`
     - Logs: `/otlp/v1/logs`
   </Message>

4. Start the OpenTelemetry Collector:
   ```bash
   otelcol --config otel-collector-config.yaml
   ```

5. Configure your applications to send metrics and logs to the collector:
   - **gRPC endpoint**: `localhost:4317`
   - **HTTP endpoint**: `localhost:4318`

   Your applications can send both metrics and logs to the same collector endpoints, which will forward them to the appropriate Cockpit endpoints based on the pipeline configuration.

## Using Grafana Alloy

Grafana Alloy provides a flexible way to collect, transform, and forward metrics to Cockpit with built-in scraping capabilities.

<Message type="note">
  For this guide, we are using Grafana Alloy on macOS. Refer to the [official Grafana documentation](https://grafana.com/docs/alloy/latest/get-started/) to find the commands for your operating system.
</Message>

> ### Configuration

Reviewer 2: ```suggestion
```

Reviewer 2: This is the only sub-heading in this section, we can remove it.

1. Create a folder named `config` on your local machine.

> 2. Create a file named `cockpit.alloy` inside the `config` folder.

Reviewer 2: ```suggestion
2. Inside the `config` folder, create a file named `cockpit.alloy`.
```

   //         headers = {
   //             "X-TOKEN" = "COCKPIT_TOKEN_SECRET_KEY",
   //         }
   //     }
   // }
   ```

4. Replace the following values:
   - `https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.metrics.cockpit.fr-par.scw.cloud/otlp/v1/metrics` with your metrics endpoint URL from the [Data sources tab](https://console.scaleway.com/cockpit/dataSource)
   - `COCKPIT_TOKEN_SECRET_KEY` with your Cockpit token's secret key
   - (Optional) Uncomment and configure the logs exporter with your logs endpoint: `https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.logs.cockpit.fr-par.scw.cloud/otlp/v1/logs`

   <Message type="important">
     Make sure that your token and your data sources (metrics and/or logs) were both created in the same region.
   </Message>

   <Message type="note">
     Grafana Alloy can send both metrics and logs to Cockpit. The example above shows metrics configuration using Prometheus remote_write. To send logs, uncomment the OTLP HTTP exporter configuration and configure your log collection components to forward to it.
     - Metrics endpoint: `/otlp/v1/metrics`
     - Logs endpoint: `/otlp/v1/logs`
   </Message>

> 5. Open a terminal and [start Alloy](https://grafana.com/docs/alloy/latest/get-started/run/):

Reviewer 2: ```suggestion
5. Open a terminal and [start Alloy](https://grafana.com/docs/alloy/latest/get-started/run/). For example, on macOS:
```

   ```bash
   brew services start alloy
   ```

6. In the same terminal, run the following command:
   ```bash
   alloy run config/cockpit.alloy
   ```

   The last lines of your output should look similar to the following:
   ```
    ts=2026-01-23T14:23:05.542081Z level=info msg="now listening for http traffic" service=http addr=127.0.0.1:12345

>     ts=2026-01-23T14:23:13.432896Z level=info msg="Done replaying WAL" component_path=/ component_id=prometheus.remote_write.cockpit subcomponent=rw remote_name=aa0551 url=https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.metrics.cockpit.fr-par.scw.cloud/otlp/v1/metrics duration=12.900498616s

Reviewer: ```suggestion
   ts=2026-01-23T14:23:05.542081Z level=info msg="now listening for http traffic" service=http addr=127.0.0.1:12345
   ts=2026-01-23T14:23:13.432896Z level=info msg="Done replaying WAL" component_path=/ component_id=prometheus.remote_write.cockpit subcomponent=rw remote_name=aa0551 url=https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.metrics.cockpit.fr-par.scw.cloud/otlp/v1/metrics duration=12.900498616s
```

   ```

> ## Visualizing Your Metrics

Reviewer 2: ```suggestion
## Visualizing your metrics
```

1. Click **Cockpit** in the **Monitoring** section of the [console](https://console.scaleway.com/) side menu. The **Cockpit** overview page displays.

> 2. Click **Open dashboards** to open your preconfigured dashboards in Grafana. You are redirected to the Grafana website.

Reviewer 2: I haven't found an **Open dashboards** button. It's rather perhaps the **Access Grafana** button, right?

Author: Nice catch, it is Access Grafana

3. Log in to your Grafana account. The Grafana overview page displays.

4. Click the Grafana icon in the top left side of your screen to open the menu.

5. Click **Dashboards**, then click **New** in the top right corner of your screen. A drop-down displays.

6. Click **Import** to import a dashboard.

7. In the **Import via grafana.com** field, enter `1860`, then click **Load**.

   <Message type="note">
     `1860` is the ID of the [Node Exporter Full dashboard](https://grafana.com/grafana/dashboards/1860-node-exporter-full/) provided by Grafana.
   </Message>

8. Enter a name for your dashboard or leave the default one.

9. In the **Folder** field, select a folder in which to store your dashboard.

> 10. In the **Prometheus** field, select your metrics data source in the drop-down.

Reviewer 2: ```suggestion
10. In the **Prometheus** field, select your metrics data source from the drop-down.
```

11. Click **Import**. Your dashboard should load and display.

12. In the drop-down next to **Datasource**, select your metrics data source. By default, your dashboard retrieves your data source's data from the last 24 hours.

13. If your data source was created less than 24 hours ago, click the clock icon and adjust the time range to match its creation time.

Your dashboard should now display your metrics collected via OTLP.

> ## Viewing Logs

Reviewer 2: ```suggestion
## Viewing logs
```

To view logs sent to Cockpit:

1. Click **Cockpit** in the **Monitoring** section of the [console](https://console.scaleway.com/) side menu.

> 2. Click **Open dashboards** to open Grafana.

Reviewer 2: I haven't found an **Open dashboards** button. It's rather perhaps the **Access Grafana** button, right?

Author: Same as above

> 3. Navigate to **Explore** from the left menu.

Reviewer 2: ```suggestion
3. Navigate to **Explore** in the left menu.
```

> 4. Select your logs data source from the dropdown.

Reviewer 2: ```suggestion
4. Select your logs data source from the drop-down.
```
